test/e2e/utils/pod.go

```diff
@@ -136,6 +138,18 @@ func NFDMasterPod(opts ...PodSpecOption) *corev1.Pod {
 							},
 						},
 					},
+					SecurityContext: &corev1.SecurityContext{
+						Capabilities: &corev1.Capabilities{
+							Drop: []corev1.Capability{"ALL"},
+						},
+						Privileged:               &no,
+						RunAsNonRoot:             &yes,
+						ReadOnlyRootFilesystem:   &yes,
+						AllowPrivilegeEscalation: &no,
+						SeccompProfile: &corev1.SeccompProfile{
+							Type: corev1.SeccompProfileTypeRuntimeDefault,
+						},
```

Comment:
I remember adding the same in https://github.com/kubernetes-sigs/node-feature-discovery/pull/917#discussion_r994753825 but for some reason I excluded it at that time. So, when running e2e had similar issues with the security context. So, it makes sense to have this change.
/lgtm
/ok-to-test